.github/workflows/code-static-analysis.yml

```diff
@@ -30,48 +28,19 @@ jobs:
   CodeQL-Build:
     # CodeQL runs on ubuntu-latest, windows-latest, and macos-latest
     runs-on: ubuntu-latest
-    # Skip duplicate job for local branches
-    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository
+    strategy:
+      matrix:
+        language: ['javascript', 'python']
```

Comment:
I do not see this matrix variable used anywhere explicitly. Is it something recognized by any steps below implicitly?

Reply:
It's used by the CodeQL program though it's not [not very well documented and took me a while to figure out!](https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#changing-the-languages-that-are-analyzed)

That's what allows this to run in two parallel steps - one for JavaScript and one for Python as can be seen at the bottom of this PR:

![image](https://user-images.githubusercontent.com/10931297/102421912-79f04f80-3ffd-11eb-9b3a-4f96d735953d.png)

Normally it runs them together in one step.

Reply:
I knew how matrix configuration affects parallelism, I was just not aware that this one had an implicit behavior change associated with the specific `language` key. Apparently, the README of the GH repo of the action does not talk about it and I did not check the documentation elsewhere.